README.md

```diff
@@ -418,7 +418,7 @@ pipeline {
    [...]
 }
 ```
-
+Although it is mentioned on the documentation page of jenkins under section "updateGitlabCommitStatus" that the supported values by GitLab are ( pending, running, canceled, success, failed ) only. Using skipped as commit status will return HTTP 400 Bad Request because it is not supported by GitLab.
```

Comment:
I do not understand exactly what your sentences are trying to convey. They seem broken and poorly written.

Reply:
Could you revise or elaborate?

Reply:
https://www.jenkins.io/doc/pipeline/steps/gitlab-plugin/#updategitlabcommitstatus-update-the-commit-status-in-gitlab here it shows that there are 6 states and the person in the issue mentioned that error was difficult to resolve due to old and different documentation. So I thought maybe I should mention that the values mentioned there are kind of outdated. This will prevent such problems in future. But what according to you should be done?

Reply:
Please read the comments carefully again at https://github.com/jenkinsci/gitlab-plugin/issues/1411#issuecomment-1437952263. It has agreed upon the documentation is to be more fully explained, because currently it is barely sufficient to someone with some knowledge of Jenkins. So I do not think adding the ambiguous line you suggested would resolve the issue at all.

Reply:
Your statement that `Although it is mentioned on the documentation page of jenkins under section "updateGitlabCommitStatus" that the supported values by GitLab are (pending, running, canceled, success, failed) only` is simply inaccurate, we actually support 6 build states: pending, running, canceled, success, failed, skipped.

Reply:
Sorry but I mentioned supported by Gitlab and Jenkins because I noticed in the file that jenkins does infact support 6 values but that is not true on the side of gitlab according to me. Or what other reason would be there to have 400 error for skipped status?

Reply:
> Sorry but I mentioned supported by Gitlab and Jenkins because I noticed in the file that jenkins does infact support 6 values but that is not true on the side of gitlab according to me. Or what other reason would be there to have 400 error for skipped status?

The build states are for Jenkins, they do not apply to GitLab at all. There seems to be some confusion in your statement.

Reply:
The `400` HTTP response was due to incorrect syntax used for the declarative pipeline job.

Reply:
Yes, I can agree with you about that. But why do all the other were working fine as it was stated in the https://github.com/jenkinsci/gitlab-plugin/issues/1411#issuecomment-1437952263 ?

Reply:
I am sorry @mukul-kr Could you clarify what you mean? What do you mean by "all the other were working fine"? To whom, regarding what process(es), and in what context?

Reply:
Well that time I meant that how in that comment they have mentioned everything was working fine but Now I have executed and know what they were doing wrong which caused the error. Maybe in few hours I will have a working solution for this issue. And sorry for the confusion I have created here😞.

Reply:
Thanks very much for your contribution!

Reply:
Hey @krisstern I tried various things but nothing worked for me. So I searched a lot and found this on [gitlab docs](https://docs.gitlab.com/ee/api/commits.html#set-the-pipeline-status-of-a-commit). I think there they have mentioned that "gitlab" only supports 5 states 
![image](https://user-images.githubusercontent.com/74813712/229850330-010efbc4-d4a2-4b95-a31e-b70b8cdf5410.png)
And here in the "gitlab-plugin" we have [6 states](https://github.com/jenkinsci/gitlab-plugin/blob/dc36630454a06b8deb337f8e965bb9d560e799b6/src/main/java/com/dabsquared/gitlabjenkins/gitlab/api/model/BuildState.java). So please further guide me🥺.

Reply:
I think now this may need api changes but please give your opinion on this. 🤔

Reply:
It should be obvious that the GitLab docs have yet to be updated. And, no API changes are required on our side. If you have too much problems with this issue maybe you are not yet ready to tackle this. I suggest you try another issue in the Jenkins project to work on.

Reply:
Okay I will try something else but I have check blame on the required files and the last update on plugin status was about 5 years or so. Even for the test you have given they were updated 5 years ago. But if you find something that works can you please ping me.

Reply:
I don't understand. Just because something is updated 4 years ago does not mean it no longer works. How did you test BTW? I have no idea what you have tried to see what is wrong.

Reply:
Yes I think I am wrong here. I haven't tried much but in my local jenkins I have tried various ways that can be used to set skip status. Well I am extremely sorry for the inconvenience I have caused.

Reply:
> I don't understand. Just because something is updated 4 years ago does not mean it no longer works. How did you test BTW? I have no idea what you have tried to see what is wrong.

Well logs show that 
![image](https://user-images.githubusercontent.com/74813712/230037118-4ec32991-8afd-45b1-ab12-d5247a868dbc.png)
And in the [gitlab commit status](https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/api/commit_statuses.rb#L169-L181) "code" not "docs" there are 5 states which they accept.
Further on lines [182-183](https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/api/commit_statuses.rb#L182-L183) it is clearly shown that any other status will raise 400 API error. 
This was why I stated that we should have some API changes. But I have may have got the wrong interpretation of the problem so I may be wrong.

Reply:
Are we talking about the Jenkins GitLab plugin build states or are we talking about the matching GitLab job states? I have already gave you the reference for the tests with the skipped build states that are currently passing. What else do you need as evidence? Also, you never provided details of what you have tried. Could you tell us what you actually tried instead?

Reply:
Please note this is documentation modification that is required for resolving the issue, not code-level ones.

Reply:
> Are we talking about the Jenkins GitLab plugin build states or are we talking about the matching GitLab job states? I have already gave you the reference for the tests with the skipped build states that are currently passing. What else do you need as evidence? Also, you never provided details of what you have tried. Could you tell us what you actually tried instead? 

I am talking about the Gitlab Job states. The api call here according to me is making request there and they have that code which handles the api call. It is meant for the external jobs and not the internal gitlab ci jobs.

Reply:
> Please note this is documentation modification that is required for resolving the issue, not code-level ones. 

Thank you and I know this. I am extremely sorry that none of my solutions worked. It is just that in the process I discovered a few things which I have proposed.

Reply:
We are not responsible for the codebase on the GitLab side. I will close this pull request for now, as we are going off topic.

Reply:
Okay, thank you for your time in reviewing this.